internal/cmd/config/config_test.go

```diff
@@ -0,0 +1,247 @@
+package config_test
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/influxdata/influx-cli/v2/internal/api"
+	"github.com/influxdata/influx-cli/v2/internal/cmd"
+	"github.com/influxdata/influx-cli/v2/internal/cmd/config"
+	iconfig "github.com/influxdata/influx-cli/v2/internal/config"
+	"github.com/influxdata/influx-cli/v2/internal/mock"
+	"github.com/influxdata/influx-cli/v2/internal/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_SwitchActive(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	stdio := mock.NewMockStdIO(ctrl)
+	writtenBytes := bytes.Buffer{}
+	stdio.EXPECT().Write(gomock.Any()).DoAndReturn(writtenBytes.Write).AnyTimes()
+
+	name := "foo"
+	cfg := iconfig.Config{
+		Name:   name,
+		Active: true,
+		Host:   "http://localhost:8086",
+		Token:  "supersecret",
+		Org:    "me",
+	}
+	svc := mock.NewMockConfigService(ctrl)
+	svc.EXPECT().SwitchActive(gomock.Eq(name)).Return(cfg, nil)
+
+	cli := config.Client{CLI: cmd.CLI{ConfigService: svc, StdIO: stdio}}
+	require.NoError(t, cli.SwitchActive(name))
+	testutils.MatchLines(t, []string{
+		`Active\s+Name\s+URL\s+Org`,
+		fmt.Sprintf(`\*\s+%s\s+%s\s+%s`, cfg.Name, cfg.Host, cfg.Org),
+	}, strings.Split(writtenBytes.String(), "\n"))
+}
+
+func TestClient_PrintActive(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	stdio := mock.NewMockStdIO(ctrl)
+	writtenBytes := bytes.Buffer{}
+	stdio.EXPECT().Write(gomock.Any()).DoAndReturn(writtenBytes.Write).AnyTimes()
+
+	cfg := iconfig.Config{
+		Name:   "foo",
+		Active: true,
+		Host:   "http://localhost:8086",
+		Token:  "supersecret",
+		Org:    "me",
+	}
+	svc := mock.NewMockConfigService(ctrl)
+	svc.EXPECT().Active().Return(cfg, nil)
+
+	cli := config.Client{CLI: cmd.CLI{ConfigService: svc, StdIO: stdio}}
+	require.NoError(t, cli.PrintActive())
+	testutils.MatchLines(t, []string{
+		`Active\s+Name\s+URL\s+Org`,
+		fmt.Sprintf(`\*\s+%s\s+%s\s+%s`, cfg.Name, cfg.Host, cfg.Org),
+	}, strings.Split(writtenBytes.String(), "\n"))
+}
+
+func TestClient_Create(t *testing.T) {
```

Comment:
These tests feel a bit too coupled to the internal implementation of the client - if we're going to have real integration tests I think we could just drop these, I doubt they would save more time catching bugs than they would cost updating them if we have to make changes.

Reply:
Will keep them for now since they're the only tests we've got, but agreed that these aren't very useful. I don't expect anything in here will change drastically before the 2.1.0 release so we should (hopefully) avoid any annoyance around changing things until after that point.